src/pudl/transform/ferc1.py

```diff
@@ -3866,3 +3885,474 @@ def plants_steam_ferc1(
+    meta_converted: dict,
+    df: pd.DataFrame,
+    xbrl_factoid_name: str,
+    table_name: str,
+):
+    """Ensure all of the names in the renamed calculations are present in the df.
+
+    Add a ``source_table`` label to the calcuaiton components when a calculation
+    component shows up in another table. Log a warning if there are any missing names.
+    """
+    xbrl_types_in_calcs = set(
+        [
+            calc_part["name"]
+            for feild_info in meta_converted[table_name].values()
+            if feild_info.get("calcs")
+            for calc_part in feild_info["calcs"]
+        ]
+        + list(meta_converted[table_name].keys())
+    )
+    # logger.info(f"{xbrl_types_in_calcs=}")
+    missing_from_tbl = {
+        xbrl_type
+        for xbrl_type in xbrl_types_in_calcs
+        if xbrl_type not in df[xbrl_factoid_name].unique()
+    }
+    # logger.info(missing_from_tbl)
+    # build a lil dictionary of missing col name to source_table name
+    missing_col_to_table = {
+        calc_comp["name"]: table_name
+        for (table_name, fields) in meta_converted.items()
+        for field_info in fields.values()
+        for calc_comp in field_info.get("calcs", [])
+        if calc_comp["name"] in missing_from_tbl
+    }
+    # add an element to the calculations components to indicate when a records if from
+    # another table
+    # This could be cleaned up.. I didn't know how to do this in a dict comp way that
+    # edited a deep subcomponent of the dict.
+    # Also: this should probabyl be done in rename_calcuations_xbrl_meta
+    for field_info in meta_converted[table_name].values():
+        if field_info.get("calcs", False):
+            for calc_component in field_info.get("calcs"):
+                if calc_component["name"] in missing_col_to_table.keys():
+                    calc_component["source_table"] = missing_col_to_table[
+                        calc_component["name"]
+                    ]
+    missing_columns = [
+        col for col in missing_from_tbl if col not in missing_col_to_table.keys()
+    ]
+    if missing_col_to_table:
+        logger.info(
+            f"{len(missing_col_to_table)} columns from {table_name} that show up in "
+            "calculated fields have been labeled with 'source_table' in the metadata."
+        )
+    if missing_columns:
+        logger.warning(
+            # raise AssertionError(
+            f"{table_name}: All renamed types were not found "
+            f"in the transformed table. Missing types: {missing_columns}"
+        )
+    return meta_converted
+
+
+def check_table_calcs(
+    table_name: str,
+    table_df: pd.DataFrame,
+    dollar_value_col: str,
+    meta_converted: dict,
+):
+    """Check how well a table's calculated values are calculated."""
+    xbrl_factoid_name = FERC1_TFR_CLASSES[table_name]().params.merge_xbrl_metadata.on
+    pks = (
+        pudl.metadata.classes.Package.from_resource_ids()
```

Comment:
We can use `pudl.metadata.classes.Resource.from_id(table_name).schema.primary_key` here rather than needing to instantiate the whole `Package`